tests/sleepq_abort.c

```diff
@@ -0,0 +1,118 @@
+#include <sleepq.h>
+#include <runq.h>
+#include <ktest.h>
+#include <sched.h>
+
+#define T 6
+
+#define RAND_COUNT 13
+/* obtained with totally fair d6 rolls (1-3 true; 4-6 false) */
+static bool kinda_random_values[RAND_COUNT] = {true,  true,  false, true, false,
+                                               false, true,  false, true, true,
+                                               false, false, false};
+/* some random ordering for waiters (values from 0 to T-1) */
+static int waiters_ord[T] = {5, 2, 1, 3, 4, 0};
+static thread_t *waiters[T];
+static thread_t *waker;
+/* just to have some unused waiting channel */
+static int some_val;
+static volatile int woken_gracefully;
+static volatile int interrupted;
+
```

Comment:
Why do you need these `interrupted` and `woken_gracefully`? You don't check their value.

Reply:
Hmm, I had some idea to check them at some point. Gonna revisit that. Same for `aborted` below.

Reply:
Yeah, now (in a moment on the repo branch) they are used indeed.